resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/HttpClientUtil.scala

```diff
@@ -42,6 +48,49 @@ private[spark] object HttpClientUtil {
     Option.apply(trustContext).foreach(context => {
       httpClientBuilder = httpClientBuilder.sslSocketFactory(sslSocketFactory, context)
     })
+    val uriObjects = uris.map(URI.create)
+    val httpUris = uriObjects.filter(uri => uri.getScheme == "http")
+    val httpsUris = uriObjects.filter(uri => uri.getScheme == "https")
+    val maybeAllProxy = Option.apply(System.getProperty(Config.KUBERNETES_ALL_PROXY))
```

Comment:
Ok that makes sense -- want to grab the system properties at the time the client is created, not the first time the class is accessed (which is less predictable)